### ISSUE TITLE
OnRowOrError() returns "json: cannot unmarshal object into Go struct field DataSetCompletion.OneApiErrors of type string"

### ISSUE THREAD

**Issue by liranms** [Azure/azure-kusto-go#182]
Reopening the original issue #124
I have a log for the reproduction of the issue - it also worth fixing the "%v" printing of the json.RawMessage
@AsafMah @yogilad 

```
'json parsing failed: [123 34 70 114 97 109 101 84 121 112 101 34 58 34 68 97 116 97 83 101 116 67 111 109 112 108 101 116 105 111 110 34 44 34 72 97 115 69 114 114 111 114 115 34 58 116 114 117 101 44 34 67 97 110 99 101 108 108 101 100 34 58 102 97 108 115 101 44 34 79 110 101 65 112 105 69 114 114 111 114 115 34 58 91 123 34 101 114 114 111 114 34 58 123 34 99 111 100 101 34 58 34 76  ... (see below the unmarshalling) .... 114 109 97 110 101 110 116 34 58 102 97 108 115 101 125 125 93 125]';'json: cannot unmarshal object into Go struct field DataSetCompletion.OneApiErrors of type string';
```

the json that is printed there as byte array is:

```{
    "FrameType": "DataSetCompletion",
    "HasErrors": true,
    "Cancelled": false,
    "OneApiErrors": [
        {
            "error": {
                "code": "LimitsExceeded",
                "message": "Request is invalid and cannot be executed.",
                "@type": "Kusto.Data.Exceptions.KustoServicePartialQueryFailureLimitsExceededException",
                "@message": "Query execution has exceeded the allowed limits (80DA0003): The results of this query exceed the set limit of 64 MB, so not all records were returned (E_QUERY_RESULT_SET_TOO_LARGE, 0x80DA0003). See https://aka.ms/kustoquerylimits for more information and possible solutions..",
                "@context": {
                    "timestamp": "2023-04-23T12:44:59.8860252Z",
                    "serviceAlias": "FOO",
                    "machineName": "KENGINE000001",
                    "processName": "Kusto.WinSvc.Svc",
                    "processId": 2984,
                    "threadId": 4716,
                    "clientRequestId": "KGC.execute;aaaaaaaa-6fda-43da-a536-6fdf7bf82491",
                    "activityId": "aaaaaaaa-ad00-436c-a8f6-93e2f97b67ac",
                    "subActivityId": "aaaaaaaa-ad00-436c-a8f6-93e2f97b67ac",
                    "activityType": "GW.Http.CallContext",
                    "parentActivityId": "aaaaaaaa-ad00-436c-a8f6-93e2f97b67ac",
                    "activityStack": "(Activity stack: CRID=KGC.execute;aaaaaaae-6fda-43da-a536-6fdf7bf82491 ARID=507e15c1-ad00-436c-a8f6-93e2f97b67ac > GW.Http.CallContext/aaaaaaaa-ad00-436c-a8f6-93e2f97b67ac)"
                },
                "@permanent": false
            }
        }
    ]
}
```

Looks like the problem is that `OneAPIErrors` field (in `DataSetCompletion`) is expected to be a `[]string`, where it is actually a list of jsons
consider changing it to `[]any`

_Originally posted by @liranms in https://github.com/Azure/azure-kusto-go/issues/124#issuecomment-1519061201_

**Comment by AsafMah** [comment]
Thank you!
I missed your comment after the issue was closed
Fixed in #183, I'll release a version soon

**Comment by AsafMah** [comment]
Fixed in v0.12.1 - please re-open if you still encounter this.
